### PR TITLE
adapt to nitro instances such as i3.metal and c5.9xlarge

### PR DESCRIPTION
**Pull request by boris-dimitrov**
also support python3.6+ weird binary subprocess.command_output

### CODE REVIEW

**Comment by boris-dimitrov** [comment]
I manually tested some parts of it but also should test in the cloud init script for ASGs by specifying a branch install for s3mi, before this is merged into prod.

The python 3 binary output of subprocess.command_output -- that almost negates the win from f-strings.

File: README.md [hunk 1]
```diff
@@ -53,12 +53,12 @@ Pronounced *semi*.
 
   * Example:
 
-      `s3mi raid my_raid 7 214`
+      `s3mi raid my_raid 3 668`
```
Comment (boris-dimitrov) [inline README.md:56]: The change here represents a better trade off between IOPS and bandwidth for same cost.

File: README.md [hunk 2]
```diff
@@ -76,34 +76,23 @@ Pronounced *semi*.
...
-  * Design question:  What if the instance where the RAID array
-    is to be mounted shouldn't have permissions to create new EBS
-    volumes?
 
 ## `s3mi raid array-name <block_device> <block_device> ...`
 
   Use RAID 0 with **instance NVME devices**.  For example,
 
-  `s3mi raid my_raid /dev/nvme{0..7}n1`
+  `s3mi raid my_raid /dev/nvme{1..8}n1`
```
Comment (boris-dimitrov) [inline README.md:90]: This is the new device numbering for nitro instances such as i3.metal.  The old numbering is still supported.

File: scripts/s3mi [hunk 3]
```diff
@@ -77,10 +77,15 @@ def safe_remove(f):
         pass
 
 
+def check_output(command):
+    # Compatibility with both 2.7 and 3.7 requires this bizarre thing.
+    return subprocess.check_output(command).decode()
```
Comment (boris-dimitrov) [inline scripts/s3mi:82]: I also tested this under 2.7 btw.

File: scripts/s3mi [hunk 4]
```diff
@@ -290,12 +297,13 @@ def append(part, baton):
     with open(part, 'rb') as pf:
         segment_bytes = pf.read()
     baton.acquire()
-    sys.stdout.write(segment_bytes)
+    bytes_written = os.write(sys.stdout.fileno(), segment_bytes)
```
Comment (boris-dimitrov) [inline scripts/s3mi:300]: @jshoe

**Review by jshoe — APPROVED** [review]

**Comment by boris-dimitrov** [comment]
Completed tests on staging last night.   All good.  Merging into master will deploy this to prod.

**Pull request merged**